### PR TITLE
[UI] Use Plotly as artifact image viewer to make it interactive

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?
Use Plotly as artifact image viewer, which allows zooming and panning ([plotly doc](https://plot.ly/python/images/#zoom-on-static-images)).

image-viewer
 

## How is this patch tested?
Manually tested
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
### What component(s) does this PR affect?

- [x] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
I came up with this feature when I saw a chart with very small tick labels. For the current UI, we have to download the image, open it with an image viewer, and zoom it, which is a bit of a pain for me.

**Comment by smurching** [comment]
@harupy thanks for the PR, this looks pretty solid - let us know when it's ready for review (by unmarking the PR as a draft) :)

**Comment by harupy** [comment]
@smurching Thanks for the comment! I will!

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js [hunk 1]
```diff
@@ -2,18 +2,97 @@ import React, { Component } from 'react';
...
+          {
+            <Plot
+              data={[
+                {
+                  x: [0, width],
+                  y: [0, height],
+                  type: 'scatter',
+                  mode: 'markers',
+                  marker: { opacity: 0, size: 0 },
+                  hoverinfo: 'none',
```
Comment (harupy) [inline mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js]: A dummy scatter to make the auto range work

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js [hunk 2]
```diff
@@ -2,18 +2,97 @@ import React, { Component } from 'react';
...
+                },
+              ]}
+              layout={{
+                width: width * (500 / height),
+                height: 500,
+                xaxis: { visible: false, autorange: true },
+                yaxis: { visible: false, autorange: true, scaleanchor: 'x', scaleratio: 1 },
```
Comment (harupy) [inline mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js]: `scaleanchor: 'x', scaleratio: 1` equalizes the scales of x-axis and y-axis ([plotly doc](https://plot.ly/python/axes/#fixed-ratio-axes)).

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js [hunk 3]
```diff
@@ -2,18 +2,97 @@ import React, { Component } from 'react';
...
+    img.onload = () => this.setState({ width: img.width, height: img.height });
+  };
+
+  componentDidMount = () => {
+    this.resize();
+  };
+
+  componentDidUpdate = prevProps => {
+    if (prevProps.path !== this.props.path) {
+      this.resize();
```
Comment (harupy) [inline mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js]: Resize the figure only when a new image is selected.

**Review by smurching — APPROVED** [review]
@harupy LGTM, thanks for the PR!

**Comment by harupy** [comment]
@smurching 
Do I have to push a commit to run the Travis build?

File: mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js [hunk 4]
```diff
@@ -2,18 +2,97 @@ import React, { Component } from 'react';
...
+                  x: [0, width],
+                  y: [0, height],
+                  type: 'scatter',
+                  mode: 'markers',
+                  marker: { opacity: 0, size: 0 },
+                  hoverinfo: 'none',
+                },
+              ]}
+              layout={{
+                width: width * (500 / height),
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js]: There might a better way to adjust the figure size without hardcoding width and height.

**Comment by harupy** [comment]
@smurching Never mind my comment above.

**Comment by harupy** [comment]
@smurching Thanks for merging! When will the next version be released?

**Comment by harupy** [comment]
btw the dummy trace can be removed. I tested the following code on [Binder](https://mybinder.org/v2/gh/plotly/plotly.py-docs/master?filepath=python/images.md) and it worked.

```diff
import plotly.graph_objects as go

# Create figure
fig = go.Figure()

# Constants
img_width = 1600
img_height = 900
scale_factor = 0.5

- Add invisible scatter trace.
- This trace is added to help the autoresize logic work.
- fig.add_trace(
-     go.Scatter(
-         x=[0, img_width * scale_factor],
-         y=[0, img_height * scale_factor],
-         mode="markers",
-         marker_opacity=0
-     )
- )

# Configure axes
fig.update_xaxes(
    visible=False,
    range=[0, img_width * scale_factor]
)

fig.update_yaxes(
    visible=False,
    range=[0, img_height * scale_factor],
    # the scaleanchor attribute ensures that the aspect ratio stays constant
    scaleanchor="x"
)

# Add image
fig.update_layout(
    images=[go.layout.Image(
        x=0,
        sizex=img_width * scale_factor,
        y=img_height * scale_factor,
        sizey=img_height * scale_factor,
        xref="x",
        yref="y",
        opacity=1.0,
        layer="below",
        sizing="stretch",
        source="https://raw.githubusercontent.com/michaelbabyn/plot_data/master/bridge.jpg")]
)

# Configure other layout
fig.update_layout(
    width=img_width * scale_factor,
    height=img_height * scale_factor,
    margin={"l": 0, "r": 0, "t": 0, "b": 0},
)

# Disable the autosize on double click because it adds unwanted margins around the image
# More detail: https://plot.ly/python/configuration-options/
fig.show(config={'doubleClick': 'reset'})
```

**Comment by smurching** [comment]
@harupy thank you for the PR! We'll likely do a 1.4.0 release early next week during Pacific time business hours (e.g. maybe next Tuesday)

**Comment by harupy** [comment]
@smurching Great! MLflow integrated in Databricks also will be updated ?

**Pull request merged**